packages/node/src/models/proposed-app-instance-info.ts

```diff
@@ -74,6 +81,48 @@ export class ProposedAppInstanceInfo implements AppInstanceInfo {
     this.respondingAddress = proposeParams.respondingAddress;
     this.initialState = proposeParams.initialState;
     this.intermediaries = proposeParams.intermediaries;
+    this.id = overrideId || this.getIdentityHashFor(channel!);
```

Comment:
How does getting an identity hash for a channel relate to an ID for a proposed AppInstance?

Reply:
The id of a proposed app instance is now the identity hash of the app instance it is proposed to become.

Reply:
No I get that - I'm saying how is that computed based solely on the channel and no information from the proposal? (at least according to the `getIdentityHashFor` fn signature)

Reply:
it is a method on the class, so it does use the params because the object has those values set